tests/python/test_multidevice.py

```diff
@@ -111,6 +111,70 @@ def multidevice_schedule(self):
     )
 
 
+@pytest.mark.mpi
+def test_linear_loop_split(mpi_test):
+    class Model(FusionDefinition):
+        def __init__(self, num_devices, batch, sequence, hidden):
+            super().__init__()
+            self._num_devices = num_devices
+            self._batch = batch
+            self._sequence = sequence
+            self._hidden = hidden
+
+        def definition(self):
+            d, b, s, e = self._num_devices, self._batch, self._sequence, self._hidden
+            self.inp = self.define_tensor([b, s, e])
+            self.weight = self.define_tensor([d * e, e], contiguity=[True, True])
+            self.bias = self.define_tensor([d * e], contiguity=[True])
```

Comment:
```suggestion
            self.weight = self.define_tensor([d * e, e], contiguity=True)
            self.bias = self.define_tensor([d * e], contiguity=True)
```
Also, a question for you: is contiguity=True necessary here? Is there a problem with non-contiguous inputs?

Reply:
No, it is not necessary.
Thunder-generated definitions always specify this flag, so I tend to specify this flag in manual definitions as well.